jobs/orcid/RevokeOrcidToken.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/**
+ * @file jobs/orcid/RevokeOrcidToken.php
+ *
+ * Copyright (c) 2014-2024 Simon Fraser University
+ * Copyright (c) 2000-2024 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class DepositOrcidSubmission
+ *
+ * @ingroup jobs
+ *
+ * @brief Job to revoke a user's ORCID access token for the application.
+ */
+
+namespace pkp\jobs\orcid;
+
+use APP\core\Application;
+use GuzzleHttp\Exception\ClientException;
+use PKP\context\Context;
+use PKP\identity\Identity;
+use PKP\jobs\BaseJob;
+use PKP\orcid\OrcidManager;
+
+class RevokeOrcidToken extends BaseJob
+{
+    public function __construct(
+        private readonly Context  $context,
+        private readonly Identity $user
+    ) {
+        parent::__construct();
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function handle()
+    {
+        $token = $this->user->getData('orcidAccessToken');
+        $httpClient = Application::get()->getHttpClient();
+        $headers = ['Accept' => 'application/json'];
+
+        $postData = [
+            'token' => $token,
+            'client_id' => OrcidManager::getClientId($this->context),
+            'client_secret' => OrcidManager::getClientSecret($this->context)
+        ];
+
+        try {
+            $httpClient->request(
+                'POST',
+                OrcidManager::getTokenRevocationUrl(),
+                [
+                    'headers' => $headers,
+                    'form_params' => $postData,
+                ],
+            );
+
+            OrcidManager::logInfo('Token revoked for user, with ID: ' . $this->user->getId());
```

Comment:
@taslangraham, everything else looks good, but I just noticed this. It could be either an author or a user, so it would be helpful to have the log reflect whichever type it is.

Reply:
updated